### ISSUE TITLE
On the Windows app SDK using CameraHelper, FrameArrived time execution for a period of time will not be triggered.

### ISSUE THREAD

**Issue by GreenShadeZhang** [CommunityToolkit/Windows#375]
### Describe the bug

When I use wasdk 1.1.2, the processing of camera frame data, at first the FrameArrived event is normal, but after a period of time, the event does not trigger, I suspect it is was wasdk bug, but I will confirm it here first.
The uwp code is normal 
like this
https://github.com/GreenShadeZhang/GreenShade.UWPDemo/tree/master/GreenShade.ML.EmoticonDetection

But when the code is ported to winui, it can only be normal for a while.

https://github.com/GreenShadeZhang/GreenShade.UWPDemo/tree/master/GreenShade.WinUI.EmoticonDetection

![图片](https://user-images.githubusercontent.com/24240675/179433571-ac0fdd65-ffcb-427c-a341-43ce6a30f2b6.png)

### Reproducible in sample app?

- [ ] This bug can be reproduced in the sample app.

### Steps to reproduce

```text
1. Download the code
2. Run the relevant code
```

### Expected behavior

I want the FrameArrived event of the video stream to work fine on the Windows app SDK so I can do emoticon analysis.

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [X] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Visual Studio Version

2022

### Device form factor

Desktop

### Help us help you

No.

**Comment by GreenShadeZhang** [comment]
> I believe the issue here is that CameraHelper is calling `VideoMediaFrame.GetVideoFrame` without disposing the result. [This code](https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/6bfe8a5d21feb8818a8048b9fc41a84a70af8b0f/Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs#L297) is where the issue is occurring.
> 
> Disposing the video frame after the event handler is invoked seems to work. I’m not sure if there would be an issue in doing that for API consumers, though.
> 
> Additionally, the `Reader_FrameArrived` should check if there are any subscribers to the `CameraHelper.FrameArrived` event before getting the video frame, so the common scenario of using `CameraPreview`, which does not rely on that event, can avoid the overhead.

Yesterday I located the cause of the problem. I found that it was still a problem with the release of unmanaged objects. I will test more today to see if I can solve this problem. If so, I will file a PR.

**Comment by ThreeSevenths** [comment]
Hi Everyone, I am still encountering an issue using the latest RC build, 8.1.240606-rc. My preview freezes after 400-500 frames. From what I can tell from Source Link, I cannot dispose the `VideoFrame` in my `FrameArrived` event handler because the getter on `FrameEventArgs` makes a copy of the original. 

Can someone confirm:
1. #376 is included in 8.1.240606-rc
2. If the fix is not included, when will it be?
3. If the fix is not included, that the only way for me to get the original `VideoFrame` from `FrameEventArgs` would be to reflect out the internal field `_videoFrame` and dispose of it in my handler
4. If there is another workaround for this issue or if I am doing something wrong?

My Window code looks like this below:

MainWindow.xaml.cs
```cs
public sealed partial class MainWindow : Window
{
    private readonly SemaphoreSlim _CameraSemaphore;

    public MainWindow()
    {
        _CameraSemaphore = new SemaphoreSlim(1);

        this.InitializeComponent();

        this.Activated += MainWindow_Activated;
        this.Closed += MainWindow_Closed;
    }

    private void MainWindow_Closed(object sender, WindowEventArgs args)
    {
        this.Closed -= MainWindow_Closed;
        UnloadCamera();
    }

    private void MainWindow_Activated(object sender, WindowActivatedEventArgs args)
    {
        this.Activated -= MainWindow_Activated;
        LoadCamera();
    }

    private async void UnloadCamera()
    {
        CameraPreview.CameraHelper.FrameArrived -= CameraHelper_FrameArrived;
        await CameraPreview.CameraHelper.CleanUpAsync();
        CameraPreview.Stop();
        CameraPreview.PreviewFailed -= CameraPreview_PreviewFailed;
    }

    private async void LoadCamera()
    {
        await _CameraSemaphore.WaitAsync();

        await CameraPreview.StartAsync();
        var result = await CameraPreview.CameraHelper.InitializeAndStartCaptureAsync();
        CameraPreview.PreviewFailed += CameraPreview_PreviewFailed;
        CameraPreview.CameraHelper.FrameArrived += CameraHelper_FrameArrived;

        _CameraSemaphore.Release();

    }

    static volatile int _Count = 0;

    private void CameraHelper_FrameArrived(object sender, CommunityToolkit.WinUI.Helpers.FrameEventArgs e)
    {

        try
        {
            var frame = e.VideoFrame;

            var framenumber = Interlocked.Increment(ref _Count);

            Debug.WriteLine("Frame {0}", framenumber);
        }
        catch(Exception ex)
        {
            Debugger.Break();
        }

    }

    private void myButton_Click(object sender, RoutedEventArgs e)
    {
        myButton.Content = "Clicked";
    }

    private void CameraPreview_PreviewFailed(object sender, CommunityToolkit.WinUI.Controls.PreviewFailedEventArgs e)
    {
        Debug.WriteLine("PreviewFailed");
    }
}
```

MainWindow.xaml
```xml
<Window
    x:Class="ZXIngTest1.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:ZXIngTest1"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:controls="using:CommunityToolkit.WinUI.Controls"
    mc:Ignorable="d">

    <StackPanel Orientation="Vertical" HorizontalAlignment="Center" VerticalAlignment="Center">
        <controls:CameraPreview x:Name="CameraPreview" Margin="10" Height="320" />
        <Button x:Name="myButton" Click="myButton_Click" HorizontalAlignment="Center">Click Me</Button>
    </StackPanel>
</Window>

```
